protocol state transition timer isn't canceled on shutdown

```
panic: send on closed channel

goroutine 74 [running]:
github.com/blinklabs-io/gouroboros/protocol.(*Protocol).SendError(...)
        /go/pkg/mod/github.com/blinklabs-io/gouroboros@v0.33.2/protocol/protocol.go:141
github.com/blinklabs-io/gouroboros/protocol.(*Protocol).setState.func1()
        /go/pkg/mod/github.com/blinklabs-io/gouroboros@v0.33.2/protocol/protocol.go:392 +0xbb
created by time.goFunc
        /usr/local/go/src/time/sleep.go:176 +0x32
```

--- a/protocol/protocol.go
+++ b/protocol/protocol.go
@@ -107,6 +107,13 @@ func (p *Protocol) Start() {
 		close(p.sendStateQueueChan)
 		close(p.recvReadyChan)
 		close(p.sendReadyChan)
+		if p.stateTransitionTimer != nil {
+			// Stop timer and drain channel
+			if !p.stateTransitionTimer.Stop() {
+				<-p.stateTransitionTimer.C
+			}
+			p.stateTransitionTimer = nil
+		}
 	}()
 	// Set initial state
 	p.setState(p.config.InitialState)

